src/orion/algo/hyperband.py

```diff
@@ -296,6 +297,8 @@ def set_state(self, state_dict: dict) -> None:
         self.seed_rng(state_dict["seed"])
         self.rng.set_state(state_dict["rng_state"])
         self.trial_to_brackets = state_dict["trial_to_brackets"]
+        self.budgets = state_dict["budgets"]
+        self.brackets.clear()
         while len(self.brackets) < len(state_dict["brackets"]):
             self.append_brackets()
         assert len(self.brackets) == len(state_dict["brackets"]), "corrupted state"
```

Comment:
Is this one of the subtil bugs you referred to in the PR description?

Reply:
Yes!

Reply:
Ok great, good catch! :)